tutorials/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,176 @@
+# # State debug statistics
+#
+# This page shows how to use the ```StateCheck``` functions to get basic
+# statistics for nodal values of fields held in ClimateMachine ```MPIStateArray```
+# data structures. The ```StateCheck``` functions can be used to
+#
+# 1. Generate statistics on ```MPIStateArray``` holding the state #    of a ClimateMachine experiment.
+# 
+#    and to
+#
+# 2. Compare against saved reference statistics from ClimateMachine ```MPIStateArray```
+#    variables. This can enable simple automated regression test checks for
+#    detecting unexpected changes introduced into numerical experiments
+#    by code updates.
+#
+# These two cases are shown below:
+
+
+# ## 1. Generating statistics for a set of MPIStateArrays
+#
+# Here we create a callback that can generate statistics for an arbitrary
+# set of the MPIStateArray type variables of the sort that hold persistent state for
+# ClimateMachine models. We then invoke the call back to show the statistics.
+#
+# In regular use the ```MPIStateArray``` variables will come from model configurations.
+# Here we create a dummy set of ```MPIStateArray``` variables for use in stand alone
+# examples.
+
+# ### Create a dummy set of MPIStateArrays
+#
+# First we set up two ```MPIStateArray``` variables. This need a few packages to be in placeT,
+# and utilizes some utility functions to create the array and add named
+# persistent state variables.
+# This is usually handled automatically as part of model deefinition in regular
+# ClimateMachine activity.
+
+# Set up a basic environment
+using MPI
+using StaticArrays
+using Random
+using ClimateMachine
```

Comment:
If you add `ClimateMachine.init()` here, you don't need the `MPI.Init()` call below. Also, `ClimateMachine.array_type()`, used later on, will not return the correct type unless `ClimateMachine.init()` is called (it checks CUDA availability and `$CLIMATEMACHINE_GPU`).